frontend/src/components/core/Sidebar/styled-components.ts

```diff
@@ -318,3 +255,26 @@ export const StyledResizeHandle = styled.div(({ theme }) => ({
     backgroundImage: `linear-gradient(to right, transparent 20%, ${theme.colors.fadedText20} 28%, transparent 36%)`,
   },
 }))
+
+export interface StyledSidebarNavButtonProps {
+  isExpanded: boolean
+}
```

Comment:
Nit: Same, looks like `isExpanded` is no longer used for styling the `StyledSidebarNavButton` and the entire interface can be removed.

Reply:
@mayagbarnes Yeah, I tried removing them, but it leads to errors in the `.tsx` component (see example below):

<img width="842" alt="Screenshot 2023-02-23 at 11 27 19 AM" src="https://user-images.githubusercontent.com/103376966/220936190-f7d1b5db-1b38-4c18-a325-907a9afd24cf.png">

Should we keep them as they are? Or move them to the Typescript component maybe?

Reply:
This error is expected and is just a reminder to remove the special props you no longer need/ defined as being allowed from being passed to the `StyledSidebarNavItems` component.

**FWIW addtl context:**
Previously we explicitly defined an interface `StyledSidebarNavItemsProps` which specified that the `StyledSidebarNavItems` component has two mandatory props extending the typical allowed props - `isExpanded` & `hasSidebarElements`. We define these interfaces when we need the styled components defined in any `styled-components.ts` file to be aware of component information for styling purposes. Thus the `isExpanded` & `hasSidebarElements` props were passed through the `StyledSidebarNavItems` component in the relevant `.tsx` file to provide this info to the styled component. 

If the styled component does not need any additional information (ex: whether the sidebar nav is expanded) other than the theme for styling, then a specially defined interface is not necessary, and special props passed to the `StyledSidebarNavItems` component are not necessary/will cause the error you encountered above.

**TLDR Part 1: Your styled-components.ts file should look like:**
```javascript
// Remove this interface defining special props passed to the styled component
// export interface StyledSidebarNavItemsProps {
//   isExpanded: boolean
//   hasSidebarElements: boolean
// }

// Eliminate the special props reference
export const StyledSidebarNavItems = styled.ul(
  ({ theme }) => {
    return {
      listStyle: "none",
      margin: 0,
      paddingTop: theme.sizes.sidebarTopSpace,
    }
  }
)
```

**TLDR Part 2: Your SidebarNav.tsx file should look like:**
```javascript
<StyledSidebarNavContainer data-testid="stSidebarNav">
     // Eliminate the special props being passed to the styled component
      <StyledSidebarNavItems
        ref={navItemsRef}
      >
// ... rest here
```

Reply:
Got it, thanks for the explanation! I've fixed them all in [b61a5c3](https://github.com/streamlit/streamlit/pull/6050/commits/b61a5c34b3a4c1389b3a3d61f0bd4cb89516c827)